tofhir-server/src/main/scala/io/tofhir/server/service/project/ProjectFolderRepository.scala

```diff
@@ -167,10 +166,13 @@ class ProjectFolderRepository(config: ToFhirEngineConfig) extends IProjectReposi
    * @param schemaId
    */
   def deleteSchema(projectId: String, schemaId: String): Unit = {
-    val project: Project = projects(projectId)
-    projects.put(projectId, project.copy(schemas = project.schemas.filterNot(s => s.id.equals(schemaId))))
-
-    updateProjectsMetadata()
+    val project: Option[Project] = projects.get(projectId)
```

Comment:
This Option[Project] is not necessary after separating the project deletion logic, but I left code as modified because I thought that this way was safer.